connectivity/tests/multicast.go

```diff
@@ -0,0 +1,363 @@
+			_, stdErr, err := client.ExecInPodWithStderr(ctx, pod.Namespace, pod.Name, defaults.AgentContainerName, cmd)
+			if err != nil {
+				if !strings.Contains(stdErr.String(), notExistMsg) {
+					errMsg := fmt.Sprintf("Error: %v, Stderr: %s", err, stdErr.String())
+					errCh <- errors.New(errMsg)
+					t.Fatalf("Fatal error occurred while checking multicast group %s in %s", testMulticastGroupIP, pod.Spec.NodeName)
+					return
+				}
+				s.addNodeWithoutGroup(pod.Spec.NodeName)
+				cmd = []string{"cilium-dbg", "bpf", "multicast", "group", "add", testMulticastGroupIP}
+				_, stdErr, err := client.ExecInPodWithStderr(ctx, pod.Namespace, pod.Name, defaults.AgentContainerName, cmd)
+				if err != nil {
+					errMsg := fmt.Sprintf("Error: %v, Stderr: %s", err, stdErr.String())
+					errCh <- errors.New(errMsg)
+					t.Fatalf("Fatal error occurred while creating multicast group %s in %s", testMulticastGroupIP, pod.Spec.NodeName)
+					return
+				}
+			}
+			// Add all ciliumInternalIPs of all nodes to the multicast group as subscribers
+			for ip, podName := range ipToPodMap {
+				if ip.IP != "" && pod.Name != podName { // My node itself does not need to be in a multicast group.
+					cmd = []string{"cilium-dbg", "bpf", "multicast", "subscriber", "add", testMulticastGroupIP, ip.IP}
+					_, stdErr, err := client.ExecInPodWithStderr(ctx, pod.Namespace, pod.Name, defaults.AgentContainerName, cmd)
+					if err == nil {
+						s.addNotSubscribePodAddress(pod.Spec.NodeName, ip)
+					} else if !strings.Contains(stdErr.String(), alreadyExistMsg) {
+						errMsg := fmt.Sprintf("Error: %v, Stderr: %s", err, stdErr.String())
+						errCh <- errors.New(errMsg)
+						t.Fatalf("Fatal error occurred while adding node %s to multicast group %s in %s", ip.IP, testMulticastGroupIP, pod.Spec.NodeName)
+						return
+					}
+				}
+			}
+		}(ciliumPod)
+	}
+
+	wg.Wait()
+	close(errCh)
+
+	var errRet error
+	for fetchdata := range errCh {
+		if fetchdata != nil {
+			errRet = errors.Join(errRet, fetchdata)
+		}
+	}
+	return errRet
+}
+
+// Delete multicast group in designated node
+func (s *socatMulticast) delGroup(ctx context.Context, t *check.Test, nodeName string) error {
+	ct := t.Context()
+	client := ct.K8sClient()
+
+	ciliumPodsList, err := client.ListPods(ctx, ct.Params().CiliumNamespace, metav1.ListOptions{LabelSelector: defaults.AgentPodSelector})
+	if err != nil {
+		return err
+	}
+	ciliumPods := ciliumPodsList.Items
+
+	for _, ciliumPod := range ciliumPods {
+		if nodeName == ciliumPod.Spec.NodeName {
+			cmd := []string{"cilium-dbg", "bpf", "multicast", "group", "delete", testMulticastGroupIP}
+			_, stdErr, err := client.ExecInPodWithStderr(ctx, ciliumPod.Namespace, ciliumPod.Name, defaults.AgentContainerName, cmd)
+			if err != nil {
+				if !strings.Contains(stdErr.String(), notExistMsg) {
+					errMsg := fmt.Sprintf("Error: %v, Stderr: %s", err, stdErr.String())
```

Comment:
minor thing, can we include the testMulticastGroupIP here, might make it easier for someone to debug issues in the future.